ui/src/pages/LoginRedirect/index.tsx

```diff
@@ -0,0 +1,21 @@
+import {useEffect} from 'react';
+import {sendGoogleAccessToken} from '../../api/googleOauth';
+import {useNavigate} from 'react-router-dom';
+
+export const LoginRedirect: React.FC = () => {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    // eslint-disable-next-line node/no-unsupported-features/node-builtins
+    const access_token = new URLSearchParams(window.location.hash).get(
+      'access_token'
+    );
+
+    if (!access_token || access_token.length === 0) {
+      return navigate('/');
+    } else {
+      sendGoogleAccessToken(access_token, navigate);
```

Comment:
Allowing `sendGoogleAccessToken` to use `navigate` gives it too much control. I would create a helper method `redirectToRoot()` and send it instead.

```ts
const redirectToRoot = () => navigate('/');
...
sendGoogleAccessToken(access_token, redirectToRoot);
```